shared/src/balancer/event_handler.rs

```diff
@@ -98,20 +100,61 @@ impl PoolDataFetching for Web3 {
 /// The BalancerPool struct represents in-memory storage of all deployed Balancer Pools
 #[derive(Derivative)]
 #[derivative(Debug)]
-pub struct BalancerPools {
+pub struct PoolRegistry {
     /// Used for O(1) access to all pool_ids for a given token
     pools_by_token: HashMap<H160, HashSet<H256>>,
     /// WeightedPool data for a given PoolId
-    pools: HashMap<H256, WeightedPool>,
+    pools: HashMap<H256, RegisteredWeightedPool>,
     #[derivative(Debug = "ignore")]
     data_fetcher: Box<dyn PoolDataFetching>,
 }
 
-impl BalancerPools {
+impl PoolRegistry {
+    // Since all the fields are private, we expose helper methods to fetch relevant information
+    pub fn pools_containing_token_pair(
+        &self,
+        token_pair: TokenPair,
+    ) -> HashSet<RegisteredWeightedPool> {
+        let empty_set = HashSet::new();
+        let pools_0 = self
+            .pools_by_token
+            .get(&token_pair.get().0)
+            .unwrap_or(&empty_set);
+        let pools_1 = self
+            .pools_by_token
+            .get(&token_pair.get().1)
+            .unwrap_or(&empty_set);
+        pools_0
+            .intersection(pools_1)
+            .into_iter()
+            .map(|pool_id| {
+                self.pools
+                    .get(pool_id)
+                    .expect("failed iterating over known pools")
+                    .clone()
+            })
+            .collect::<HashSet<RegisteredWeightedPool>>()
+    }
+
+    pub fn pools_containing_token_pairs(
+        &self,
+        token_pairs: HashSet<TokenPair>,
+    ) -> HashSet<RegisteredWeightedPool> {
```

Comment:
I would like to make it clear in a documentation comment whether this returns
1. all pools that contain any of the tokens(current implementation)
2. all pools that contain all of the tokens

Reply:
I have included a doc comment on both methods `pools_containing_token_pairs`and `pools_containing_token_pair`